src/Camera.cpp

```diff
@@ -0,0 +1,124 @@
+#include "Camera.h"
+
+#include <string>
+
+#include <opencv2/core/matx.hpp>
+#include <opencv2/opencv.hpp>
+#include <utility>
+
+#include <ntcore/networktables/NetworkTableInstance.h>
+#include <ntcore/networktables/DoubleArrayTopic.h>
+#include <ntcore/networktables/IntegerTopic.h>
+
+#include "Utils.h"
+
+using namespace std;
+using namespace cv;
+using namespace nt;
+
+Camera::Camera(string& id, vector<vector<double>> matrix, vector<double> distortionCoefficents, DoubleArrayPublisher tvecOut,
```

Comment:
This function is making assumptions about the vectors passed in. Additionally, the vectors aren't really changing size. Would you consider using `std::array` instead? You could even do something rust-like and ensure all of them are specific sizes at compile time.

Here's example code from another project I've done as reference.

```C++
template<std::size_t N>
HW::Spi::ERROR HW::Spi::xfer(std::array<uint8_t, N>& input, std::array<uint8_t, N>& output, bool deassert_cs) {
    _cs.set(false);
    const auto error = HAL_SPI_TransmitReceive(this->_spi, input.data(), output.data(), N, TIMEOUT_TICKS);
    if (deassert_cs) {
        _cs.set(true);
    }
    return HAL_to_SPI_err(error);
}
```